### PR TITLE
Determining object size correctly

### LINKED ISSUE #590: Failing integ test on StreamWrapper
```
Aws\Test\Integ\S3StreamWrapperTest::testOpensStreams
Failed asserting that 0 matches expected 7.

/.../tests/Integ/S3StreamWrapperTest.php:123
```

### PR DESCRIPTION
**Pull request by mtdowling**
Closes #590 

@jeremeamia @jeskew

### CODE REVIEW

**Comment by mtdowling** [comment]
The failing integration test now passes

File: tests/S3/StreamWrapperTest.php [hunk 1]
```diff
@@ -818,4 +818,19 @@ public function testCachesReaddirs()
         $this->assertEquals(2, filesize('s3://bucket/key/' . $file2));
         closedir($r);
     }
+
+    public function testReturnsStreamSizeFromHeaders()
+    {
+        $stream = Psr7\stream_for('12345');
+        $stream = Psr7\FnStream::decorate($stream, [
+            'getSize' => function () { return null; }
```
Comment (GrahamCampbell) [inline tests/S3/StreamWrapperTest.php]: the `return null;` is uneeded isn't it?

Comment (mtdowling) [inline tests/S3/StreamWrapperTest.php]: Eh, true, but it's more explicit.

**Comment by jeremeamia** [comment]
LGTM.

**Pull request merged**